multinet/api/tests/test_upload_common.py

```diff
@@ -0,0 +1,143 @@
+from typing import List
+
+from django.contrib.auth.models import User
+import pytest
+from rest_framework.response import Response
+from rest_framework.test import APIClient
+
+from multinet.api.models.upload import Upload
+from multinet.api.models.workspace import Workspace
+from multinet.api.tests.factories import UploadFactory
+from multinet.api.tests.fuzzy import TIMESTAMP_RE
+from multinet.api.utils.workspace_permissions import WorkspacePermission
+
+
+@pytest.mark.django_db
+def test_upload_rest_retrieve(
+    workspace: Workspace,
+    user: User,
+    upload_factory: UploadFactory,
+    authenticated_api_client: APIClient,
+):
+    """Test retrieval of an upload on a workspace for a user with read access."""
+    workspace.set_user_permission(user, WorkspacePermission.reader)
+    upload: Upload = upload_factory(workspace=workspace, user=user)
+    r: Response = authenticated_api_client.get(
+        f'/api/workspaces/{workspace.name}/uploads/{upload.pk}/'
+    )
+    assert r.status_code == 200
+    assert r.data == {
+        'id': upload.pk,
+        'blob': upload.blob,
+        'user': user.username,
+        'created': TIMESTAMP_RE,
+        'modified': TIMESTAMP_RE,
+        'error_messages': upload.error_messages,
+        'data_type': upload.data_type,
+        'status': upload.status,
+        'workspace': {
+            'id': workspace.pk,
+            'name': workspace.name,
+            'created': TIMESTAMP_RE,
+            'modified': TIMESTAMP_RE,
+            'arango_db_name': workspace.arango_db_name,
+            'public': workspace.public,
+        },
```

Comment:
`workspace_re(workspace)` can be used here
```suggestion
        'workspace': workspace_re(workspace)
```

You just need to import it as well